File: backend/app/models/reports.py
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel, Field

from app.models.alerts import AlertSeverity


class ReportStatus(StrEnum):
    PENDING = "PENDING"
    STREAMING = "STREAMING"
    COMPLETE = "COMPLETE"
    FAILED = "FAILED"


class MitreTactic(BaseModel):
    tactic_id: str
    tactic_name: str
    techniques: list[str] = []


class TimelineEntry(BaseModel):
    timestamp: datetime
    event: str


class AffectedAsset(BaseModel):
    hostname: str | None = None
    ip: str | None = None
    role: str | None = None


class IRReport(BaseModel):
    id: str
    alert_ids: list[str]
    status: ReportStatus = ReportStatus.PENDING
    title: str = ""
    executive_summary: str = ""
    mitre_tactics: list[MitreTactic] = []
    affected_assets: list[AffectedAsset] = []
    confidence: int = Field(0, ge=0, le=100)
    severity: AlertSeverity = AlertSeverity.LOW
    timeline: list[TimelineEntry] = []
    recommendations: list[str] = []
    raw_stream: str = ""
    created_at: datetime = Field(default_factory=datetime.utcnow)
    completed_at: datetime | None = None

File: frontend/src/types/reports.ts
import type { AlertSeverity } from './alerts'

export type ReportStatus = 'PENDING' | 'STREAMING' | 'COMPLETE' | 'FAILED'

export interface MitreTactic {
  tactic_id: string
  tactic_name: string
  techniques: string[]
}

export interface TimelineEntry {
  timestamp: string
  event: string
}

export interface AffectedAsset {
  hostname: string | null
  ip: string | null
  role: string | null
}

export interface IRReport {
  id: string
  alert_ids: string[]
  status: ReportStatus
  title: string
  executive_summary: string
  mitre_tactics: MitreTactic[]
  affected_assets: AffectedAsset[]
  confidence: number
  severity: AlertSeverity
  timeline: TimelineEntry[]
  recommendations: string[]
  raw_stream: string
  created_at: string
  completed_at: string | null
}

export interface GenerateReportRequest {
  alert_ids: string[]
}
